packages/andromeda_protocol/src/gumball.rs

```diff
@@ -0,0 +1,109 @@
+    Buy {},
+    // Sets price, max amount per wallet, and recipient
+    SaleDetails {
+        /// The price per token.
+        price: Coin,
+        /// The amount of tokens a wallet can purchase, default is 1.
+        max_amount_per_wallet: Option<Uint128>,
+        /// The recipient of the funds if the sale met the minimum sold.
+        recipient: Recipient,
+    },
+    // Automatically switches to opposite status.
+    // True means buying is allowed and minting is halted. False means the opposite.
+    SwitchStatus {},
+    // SetContractAddress {
+    //     andromeda_cw721_contract: AndrAddress,
+    // },
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+    AndrQuery(AndromedaQuery),
+    NumberOfNFTs {},
+    SaleDetails {},
+    Status {},
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum RandQueryMsg {
+    LatestDrand {},
+    GetRandomness { round: u64 },
+}
+// #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+// pub struct BeaconInfoState {
+//     pub round: u64,
+//     pub randomness: Binary,
+//     pub worker: CanonicalAddr,
+// }
+// pub const BEACONS: Map<&[u8], BeaconInfoState> = Map::new("beacons");
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct NumberOfNFTsResponse {
+    pub number: usize,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct State {
+    /// The price of each token.
+    pub price: Coin,
+    /// The max number of tokens allowed per wallet.
+    pub max_amount_per_wallet: Uint128,
+    /// The recipient of the funds upon a sale.
+    /// Most likely the contract, but could also be the splitter contract for example.
+    pub recipient: Recipient,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct StateResponse {
+    pub state: State,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct StatusResponse {
+    pub status: bool,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct RandomResult {
+    pub randomness: Binary,
+    pub worker: Addr,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema, Default)]
+pub struct GetRandomResponse {
+    pub randomness: Binary,
+    pub worker: String,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct LatestRandomResponse {
+    pub round: u64,
+    pub randomness: Binary,
+    pub worker: String,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct LatestRoundResponse {
+    pub height: String,
+    pub result: RandomResult,
+}
```

Comment:
Can we use the types from the `terrand` dependency instead of redefining them? Also for the `RandQueryMsg` above.

Reply:
It isn't for certain that we'll be using Terrand on the long run, so I removed the responses that I didn't end up using and left LatestRandomResponse.